### PR TITLE
Update identify.py

### PR DESCRIPTION
**Pull request by sara-rn**
identify Go samples with stomped PCLNTAB magic bytes

### CODE REVIEW

File: floss/language/identify.py [hunk 1]
```diff
@@ -104,6 +115,8 @@ def is_go_bin(pe: pefile.PE) -> bool:
                         logger.info("Go binary found with version %s", get_go_version(magic))
                         return True
 
+
+
```
Comment (mr-tz) [inline floss/language/identify.py:119]:
```suggestion
```

File: floss/language/identify.py [hunk 2]
```diff
@@ -117,6 +130,19 @@ def is_go_bin(pe: pefile.PE) -> bool:
                     # just for testing
                     logger.info("Go binary found with version %s", get_go_version(magic))
                     return True
+
+    # if not found, the magic bytes are patched, search for common Go functions present in all Go samples incl. obfuscated
+    for go_function in go_functions:
+        for section in pe.sections:
```
Comment (mr-tz) [inline floss/language/identify.py:136]: - can we limit to specific sections?
- would searching in the sections first for all strings be more efficient?

Comment (williballenthin) [reply floss/language/identify.py:136]: agree with #2 here - pefile.get_data is fairly slow, so recommend looping across sections in the outer loop, and then across the strings in the inner loop.

Comment (sara-rn) [reply floss/language/identify.py:135]: I suggest searching first in .rdata, where the string runtime.main is found

File: floss/language/identify.py [hunk 3]
```diff
@@ -117,6 +129,34 @@ def is_go_bin(pe: pefile.PE) -> bool:
...
+            for go_function in go_functions:
+                if go_function in section_data:
+                    logger.info("Go binary found, function name %s", go_function)
+                    return True
+
+    for section in pe.sections:
+        for go_function in go_functions:
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
```
Comment (williballenthin) [inline floss/language/identify.py:153]: would you pull these out of the inner loop please?

File: floss/language/identify.py [hunk 4]
```diff
@@ -131,15 +131,30 @@ def is_go_bin(pe: pefile.PE) -> bool:
...
     # if not found, the magic bytes may have been patched, search for common Go functions present in all Go samples including obfuscated files
-    for go_function in go_functions:
-        for section in pe.sections:
+    # look for the .rdata section first
+    for section in pe.sections:
+        try:
+            section_name = section.Name.partition(b"\x00")[0].decode("utf-8")
+        except UnicodeDecodeError:
+            continue
+        if ".rdata" == section_name:
```
Comment (mr-tz) [inline floss/language/identify.py:140]: maybe reuse this?
https://github.com/mandiant/flare-floss/blob/152316f79b075115790b9484c5196a2429000598/floss/language/rust/extract.py#L20-L25

could move this to utils most likely

Comment (mr-tz) [reply floss/language/identify.py:140]: Could move this then to https://github.com/mandiant/flare-floss/blob/master/floss/language/utils.py

**Comment by mr-tz** [comment]
bump @sara-rn 😄

**Comment by sara-rn** [comment]
I haven't forgotten about these, shall I create a new PR for
https://github.com/mandiant/flare-floss/blob/master/floss/language/utils.py?

**Comment by mr-tz** [comment]
Yes, please or update this PR. Note that the identify code changed slightly.

**Pull request closed without merging**